src/main/java/org/opensearch/security/dlic/rest/api/AccountApiAction.java

```diff
@@ -183,47 +237,106 @@ protected void handleGet(RestChannel channel, RestRequest request, Client client
     @Override
     protected void handlePut(RestChannel channel, final RestRequest request, final Client client, final JsonNode content) throws IOException {
         final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
-        final String username = user.getName();
         final SecurityDynamicConfiguration<?> internalUser = load(CType.INTERNALUSERS, false);
 
-        if (!internalUser.exists(username)) {
-            notFound(channel, "Could not find user.");
+        if (user == null){
+            badRequestResponse(channel, "User not found.");
             return;
         }
 
-        if (!isWriteable(channel, internalUser, username)) {
+        if (!internalUser.exists(user.getName())) {
+            notFound(channel, "Could not find user.");
             return;
         }
 
-        final SecurityJsonNode securityJsonNode = new SecurityJsonNode(content);
-        final String currentPassword = content.get("current_password").asText();
-        final Hashed internalUserEntry = (Hashed) internalUser.getCEntry(username);
-        final String currentHash = internalUserEntry.getHash();
-
-        if (currentHash == null || !OpenBSDBCrypt.checkPassword(currentHash, currentPassword.toCharArray())) {
-            badRequestResponse(channel, "Could not validate your current password.");
+        if (!isWriteable(channel, internalUser, user.getName())) {
             return;
         }
 
-        // if password is set, it takes precedence over hash
-        final String password = securityJsonNode.get("password").asString();
-        final String hash;
-        if (Strings.isNullOrEmpty(password)) {
-            hash = securityJsonNode.get("hash").asString();
-        } else {
-            hash = hash(password.toCharArray());
-        }
-        if (Strings.isNullOrEmpty(hash)) {
-            badRequestResponse(channel, "Both provided password and hash cannot be null/empty.");
+        // request must have 'saved_tenant' AND/OR ('current_password' AND ('password' OR 'hash'))
+        final boolean invalidSavedTenantBody = content.get("saved_tenant") == null;
+        final boolean invalidPasswordBody = content.get("current_password") == null || (content.get("password") == null && content.get("hash") == null);
+        // bad case: no valid saved_tenant body nor password body
+        if (invalidSavedTenantBody && invalidPasswordBody){
+            badRequestResponse(channel, "Invalid request. Needs a saved_tenant AND/OR (saved_password AND (password OR hash)");
             return;
         }
 
-        internalUserEntry.setHash(hash);
+        // process PUT for 'saved_tenant'
+        if (content.get("saved_tenant") != null){
```

Comment:
can use invalidSavedTenantBody

Reply:
I didnt get you. You are simply checking if the content have saved_tenant for which you already have a boolean variable.

Reply:
modified to reuse the boolean